kernel/genarch/src/mm/as_pt.c

```diff
@@ -75,9 +75,9 @@ pte_t *ptl0_create(unsigned int flags)
 	pte_t *dst_ptl0 = (pte_t *)
 	    PA2KA(frame_alloc(PTL0_FRAMES, FRAME_LOWMEM, PTL0_SIZE - 1));
 
-	if (flags & FLAG_AS_KERNEL)
-		memsetb(dst_ptl0, PTL0_SIZE, 0);
-	else {
+	memsetb(dst_ptl0, PTL0_SIZE, 0);
+
+	if (!KERNEL_SEPARATE_PTL0 && !(flags & FLAG_AS_KERNEL)) {
```

Comment:
So what's the deal with KERNEL_SEPARATE_PTL0, why is this done differently on arm64?